plugins/CoreVisualizations/Visualizations/JqplotGraph/Evolution.php

```diff
@@ -51,42 +53,44 @@ public function beforeLoadDataTable()
+        $period = Common::getRequestVar('period', false);
+        $selector = StaticContainer::get(EvolutionPeriodSelector::class);
+
+        if ($period === 'range') {
+            $date = Common::getRequestVar('date', false);
+            $requestingPeriod = Factory::build($period, $date);
+
+            // if a larger date range is selected, then for better performance and for seeing trends better we want to use
+            // a suitable period (rather than always using for example the day range)
+            $this->requestConfig->request_parameters_to_modify['period'] = $selector->getHighestPeriodInCommon($requestingPeriod, []);
+            $this->requestConfig->request_parameters_to_modify['date'] = $requestingPeriod->getRangeString();
         }
 
         $this->config->custom_parameters['columns'] = $this->config->columns_to_display;
 
         if ($this->isComparing()) {
-            $this->config->show_limit_control = false; // since we always show the days over the period, there's no point in changing the limit
+            $this->config->show_limit_control = false; // since we always show the evolution over the period, there's no point in changing the limit
+            $this->config->show_periods = false; // the periods can't be changed as they are always fixed when comparing
 
             $requestArray = $this->request->getRequestArray();
             $requestArray = ApiRequest::getRequestArrayFromString($requestArray);
 
             $requestingPeriod = Factory::build($requestArray['period'], $requestArray['date']);
 
-            $this->requestConfig->request_parameters_to_modify['period'] = 'day';
-            $this->requestConfig->request_parameters_to_modify['date'] = $requestingPeriod->getDateStart()->toString() . ',' . $requestingPeriod->getDateEnd()->toString();
-
+            $comparisonPeriods = [];
             if (!empty($requestArray['comparePeriods'])) {
-                foreach ($requestArray['comparePeriods'] as $index => $comparePeriod) {
-                    $compareDate = $requestArray['compareDates'][$index];
-                    if (Period::isMultiplePeriod($compareDate, $comparePeriod)) {
-                        continue;
-                    }
-
-                    $comparePeriodObj = Factory::build($comparePeriod, $compareDate);
-
-                    $requestArray['comparePeriods'][$index] = 'day';
-                    $requestArray['compareDates'][$index] = $comparePeriodObj->getRangeString();
-                }
-
-                $this->requestConfig->request_parameters_to_modify['compareDates'] = $requestArray['compareDates'];
-                $this->requestConfig->request_parameters_to_modify['comparePeriods'] = $requestArray['comparePeriods'];
+                $comparisonPeriods = $selector->getComparisonPeriodObjects($requestArray['comparePeriods'], $requestArray['compareDates']);
             }
+
+            $this->requestConfig->request_parameters_to_modify = $selector->setDatePeriods(
```

Comment:
When now comparing date ranges, the evolution graph may switch from day period to week or month or year to needing to fetch less archives.
This now ensures sparklines and evolution graphs use the same periods when comparing.